daemon/seccomp_linux.go

```diff
@@ -22,18 +22,18 @@ func WithSeccomp(daemon *Daemon, c *container.Container) coci.SpecOpts {
 		var profile *specs.LinuxSeccomp
 		var err error
 
+		if c.SeccompProfile == "unconfined" {
+			return nil
+		}
 		if c.HostConfig.Privileged {
 			return nil
 		}
-
 		if !daemon.seccompEnabled {
-			if c.SeccompProfile != "" && c.SeccompProfile != "unconfined" {
+			if c.SeccompProfile != "" {
 				return fmt.Errorf("seccomp is not enabled in your kernel, cannot run a custom seccomp profile")
 			}
 			logrus.Warn("seccomp is not enabled in your kernel, running container without default profile")
 			c.SeccompProfile = "unconfined"
-		}
-		if c.SeccompProfile == "unconfined" {
 			return nil
 		}
 		if c.SeccompProfile != "" {
```

Comment:
Perhaps these nested ifs could be rewritten to a `switch`. Combined with my other comment, something like;

```go
// WithSeccomp sets the seccomp profile
func WithSeccomp(daemon *Daemon, c *container.Container) coci.SpecOpts {
	return func(ctx context.Context, _ coci.Client, _ *containers.Container, s *coci.Spec) error {
		if c.SeccompProfile == "unconfined" {
			return nil
		}
		if c.HostConfig.Privileged {
			return nil
		}
		if !daemon.seccompEnabled {
			if c.SeccompProfile != "" {
				return fmt.Errorf("seccomp is not enabled in your kernel, cannot run a custom seccomp profile")
			}
			logrus.Warn("seccomp is not enabled in your kernel, running container without default profile")
			c.SeccompProfile = "unconfined"
			return nil
		}

		var err error
		switch {
		case c.SeccompProfile != "":
			s.Linux.Seccomp, err = seccomp.LoadProfile(c.SeccompProfile, s)
		case daemon.seccompProfile != nil:
			s.Linux.Seccomp, err = seccomp.LoadProfile(string(daemon.seccompProfile), s)
		default:
			s.Linux.Seccomp, err = seccomp.GetDefaultProfile(s)
		}
		return err
	}
}
```

Reply:
That's awesome, the final result does look a lot cleaner. Thank you. 👍